### DISCUSSION TITLE
How do I check for updates in pacman and show them in my status bar

### DISCUSSION THREAD

**Discussion by ghost** [i3/i3#4658] [Q&A] [answered]
I’m using i3status and I want to check for updates like I want to get notified that there are X number of updates available how can I do that?

**Comment by Airblader** [comment, marked as answer]
`pacman` comes with `checkupdates` which can be used for this purpose. For example, `checkupdates | wc -l` will tell you the number of updates available (note that this is excluding AUR). I have this script:

```
#!/usr/bin/env bash

UPDATES=$(checkupdates | wc -l)
[[ "${UPDATES}" = "0" ]] && exit 0

echo "   ${UPDATES} "
exit 0
```

… and use it in i3blocks (not i3status) as

```
[updates]
command=$HOME/scripts/updates.sh
interval=60
signal=3
urgent=true
```

If you want to add this using i3status, you need to either periodically write this information to a file and use [`read_file`](https://i3wm.org/docs/i3status.html#_file_contents), or wrap the output of `i3status` and enrich the JSON response yourself.
